Source/player.cpp

```diff
@@ -2669,23 +2669,15 @@ void AddPlrExperience(int pnum, int lvl, int exp)
 
 	// Prevent power leveling
 	if (gbIsMultiplayer) {
-		int powerLvlCap = player._pLevel < 0 ? 0 : player._pLevel;
-		if (powerLvlCap >= 50) {
-			powerLvlCap = 50;
-		}
-		// cap to 1/20 of current levels xp
-		if (exp >= ExpLvlsTbl[powerLvlCap] / 20) {
-			exp = ExpLvlsTbl[powerLvlCap] / 20;
-		}
-		// cap to 200 * current level
-		int expCap = 200 * powerLvlCap;
-		if (exp >= expCap) {
-			exp = expCap;
-		}
+		auto clampedPlayerLevel = std::clamp(static_cast<int>(player._pLevel), 0, 50);

```

Comment:
We have our own `clamp()` for a reason ;)
```suggestion
		auto clampedPlayerLevel = clamp(static_cast<int>(player._pLevel), 0, 50);
```

Reply:
Been meaning to ask about that one, it's defined as std::clamp for c++17 which is the version devilutionx targets. Is there a platform where c++17 support is incomplete that the alternate definition handles?

Reply:
Yes, Amiga and PowerPC Linux. They are on GCC 6.2 which only has partial C++17 support. There is also a guy somewhere who insist on Slackware which has GCC 5.5, it currently works, but I'm less concerned about it.